pypdf/_page.py

```diff
@@ -920,6 +920,50 @@ def getContents(self) -> Optional[ContentStream]:  # deprecated
         deprecation_with_replacement("getContents", "get_contents", "3.0.0")
         return self.get_contents()
 
+    def replace_contents(self, content: Optional[ContentStream]) -> None:
+        """
+        Replace the page contents with the new content and nullify old objects
+        Args:
+            content : new content. if None delete the content field.
+        """
+        if isinstance(self.get(PG.CONTENTS, None), ArrayObject):
+            for o in self[PG.CONTENTS]:
+                try:
+                    self._objects[o.indirect_reference.idnum - 1] = NullObject()
+                except AttributeError:
+                    pass
+        if content is None:
+            if PG.CONTENTS not in self:
+                return
+            else:
+                self.indirect_reference.pdf._objects[
```

Comment:
Mypy complains that `self.indirect_reference` might be none here. Could that actually happen?

Reply:
Some cases for independant pages. I've added some code